coral-spark/src/main/java/com/linkedin/coral/spark/utils/RelDataTypeToHiveTypeStringConverter.java

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright 2022 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.spark.utils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeField;
+import org.apache.calcite.rel.type.RelRecordType;
+import org.apache.calcite.sql.type.ArraySqlType;
+import org.apache.calcite.sql.type.MapSqlType;
+
+
+/**
+ * Transforms a RelDataType to a Hive type string such that it is parseable and semantically correct.
+ * Some example of Hive type string for a RelDataType are as follows:
+ *
+ * Example 1:
+ * RelDataType:
+ *   struct(s1: integer, s2: varchar)
+ * Hive Type String:
+ *   struct(s1:int,s2:string)
+ *
+ * Example 2:
+ * RelDataType:
+ *   map(varchar, struct(s1: integer, s2: varchar))
+ * Hive Type String:
+ *   map(string,struct(s1:int,s2:string))
+ * Example 3:
+ * RelDataType:
+ *   array(struct(s1: integer, s2: varchar))
+ * Hive Type String:
+ *   array(struct(s1:int,s2:string))
```

Comment:
nit: could you modify the examples in the doc to have consistent spacing? for ex: 
```
* RelDataType: 
*   array(struct(s1:integer, s2:varchar))
* Hive Type String:
*   array(struct(s1:int,s2:string))
```

Reply:
This is actually intentional, Hive type format requires that there is no redundant space between tokens.

Reply:
I just started the jenkins integration test, but since it is the first time we found the argument need to be hive type, it is likely we don't have a production view using `generic_project` before.